.claude/hooks/post_tool_use.py

```
@@ -0,0 +1,51 @@
+        input_data = json.load(sys.stdin)
+        
+        # Extract session_id
+        session_id = input_data.get('session_id', 'unknown')
+        
+        # Ensure session log directory exists
+        log_dir = ensure_session_log_dir(session_id)
+        log_path = log_dir / 'post_tool_use.json'
+        
+        # Read existing log data or initialize empty list
+        if log_path.exists():
+            with open(log_path, 'r') as f:
```

Ruff (0.12.2)

Unnecessary mode argument

Remove mode argument

(UP015)